File: sdks/python/orisign-bot-sdk/examples/weather_bot.py
from orisign_bot_sdk import Bot, Markup


bot = Bot(token="BOT_TOKEN", base_url="http://localhost:4000")


@bot.command("/start")
async def start(ctx):
    await ctx.reply(
        "Use /weather <city>",
        Markup.inline_keyboard([[Markup.button("Help", callbackData="help")]]),
    )


@bot.command("/weather")
async def weather(ctx):
    await ctx.reply("Plug your weather provider here.")

File: packages/bot-sdk/src/examples/weather-bot.ts
import { createBot, Markup } from '..';

const bot = createBot({
  token: process.env.BOT_TOKEN ?? '',
  baseUrl: process.env.BOT_BASE_URL ?? 'http://localhost:4000',
});

bot.command('/start', async (ctx) => {
  await ctx.reply(
    'Use /weather <city>',
    Markup.inlineKeyboard([[Markup.button('Help', { callbackData: 'help' })]]),
  );
});

bot.command('/weather', async (ctx) => {
  await ctx.reply('Weather lookup example. Plug your provider here.');
});

bot.onAction('help', async (ctx) => {
  await ctx.reply('Send /weather Novosibirsk');
});

void bot.startPolling();
